src/app/app.component.ts

```diff
@@ -12,34 +14,28 @@ import { User } from "./shared/models/user.model";
 })
 export class AppComponent implements OnInit {
   title = "courses-app";
-  isAuthorized!: boolean;
-  isAdmin!: boolean;
+  //isAuthorized!: boolean;
+  isAdmin$: Observable<boolean> = this.userStateFacade.isAdmin$;
   userName: string | undefined = "";
-  loggedInUser!: User;
+  userName$: Observable<string | undefined> = this.userStateFacade.user$.pipe(
+    map((user) => user?.name)
+  );
+  //loggedInUser!: User;
 
   //  text for buttons
   btnTextLogin: string = ButtonLabels.login;
   btnTextLogout: string = ButtonLabels.logout;
 
   constructor(
     public authService: AuthService,
-    private userStoreService: UserStoreService
+    private userStoreService: UserStoreService,
+    private userStateFacade: UserStateFacade,
+    private authenticationFacade: AuthenticationFacade
   ) {}
 
-  ngOnInit() {
-    this.userStoreService.loggedInUser$.subscribe({
-      next: (user) => {
-        this.isAdmin = user.role === "admin" ? true : false;
-        this.userName = user.name;
-      },
-    });
-
-    this.authService.isAuthorized$.subscribe(
-      (isAuth) => (this.isAuthorized = isAuth)
-    );
-  }
+  ngOnInit() {}
```

Comment:
since this is empty, I guess then the component does not need to implement OnInit hook.